pkg/reconciler/subscription/subscription.go

```diff
@@ -335,9 +336,21 @@ func (r *Reconciler) trackAndFetchChannel(ctx context.Context, sub *v1alpha1.Sub
 // underlying channels), fetch and validate the "backing" channel.
 func (r *Reconciler) getChannel(ctx context.Context, sub *v1alpha1.Subscription) (*eventingduckv1alpha1.Channelable, pkgreconciler.Event) {
 	logging.FromContext(ctx).Warn("GETTING channel", zap.Any("channel", sub.Spec.Channel))
+
+	// let's use the v1alpha1 API for now, until we migrate this reconciler
+	// to work directly on v1beta1 object
+	channel := sub.Spec.Channel
+	if channel.GroupVersionKind().Version == "v1beta1" {
+		channel.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   channel.GroupVersionKind().Group,
+			Version: "v1alpha1",
```

Comment:
This is wrong. This will force any channel impl to be accessed via it's v1alpha1 version and that might not exist.

/hold

Reply:
Trust the subscription, and report errors back to them if it is wrong.

Reply:
I know. I might have cycles today for the proper fix.

What do you mean by trust the subscription? We don't want the patch, and the status report to fail.

Reply:
Yes, this will not work for all the cases and that's true. I don't see the subscription helping here, but when we go to compute the duck patch, it fails because v1alpha1 has an extra subscribers field there. I think getting this patch in is fine for the release, otherwise, v1beta1 channels won't work at all by the looks of things.
I think for the proper fix we'll need to know if a particular channel type supports either v1alpha1 or v1beta1 duck.
One option is to have channel implementations have more details in their CRD labels, so rather than just saying:
    messaging.knative.dev/subscribable: "true"
We could extend this to include the version number, which might be a good way to future proof this further.
Or we could rely on the versioned schema, or. But, in general we don't have enough information at this point in the code to know how to generate a patch for different subscribable ducks.
So, I think getting this in, for now seems ok and we can continue to get a better solution out today.
Thoughts?

Reply:
+1 for merging this PR  today. 

I'm working on a separate PR with a better fix by looking at the version referenced in the Subscription.